cmd/controller/server.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	prometheusMetricsServerPort            = 9402
+	prometheusMetricsServerShutdownTimeout = 10 * time.Second
+	prometheusMetricsServerReadTimeout     = 8 * time.Second
+	prometheusMetricsServerWriteTimeout    = 8 * time.Second
+	prometheusMetricsServerMaxHeaderBytes  = 1 << 20
+)
+
+type prometheusMetricsServer struct {
+	http.Server
+}
+
+func prometheusHandler() http.Handler {
+	return prometheus.Handler()
+}
+
+func newPrometheusMetricsServer() *prometheusMetricsServer {
+
+	r := mux.NewRouter()
+	r.Handle("/metrics", prometheusHandler())
+
+	// Create server and register prometheus metrics handler
+	s := &prometheusMetricsServer{
+		Server: http.Server{
+			Addr:           fmt.Sprintf(":%v", prometheusMetricsServerPort),
```

Comment:
Can we replace `prometheusMetricsServerPort` with a string that defaults to `:` or `0.0.0.0:`, instead of using a hardcoded 0.0.0.0?